modules/recoil_element.py

```diff
@@ -81,8 +81,8 @@ def __init__(self, element: Element, points: List[Point], color="red",
             self.reference_density = GlobalSettings().get_default_reference_density()
         else:
             self.reference_density = reference_density
-
-        self.manual_reference_density_checked = False
+            if reference_density == self.reference_density:
+                self.reference_density_element_default = reference_density
```

Comment:
This is dangerous because the member variable is only sometimes present. It should be always present (if needed at all), otherwise this can cause crashes when checking the value.

I would try to get rid of this new variable if possible.

It'll also cause issues when serializing/deserializing objects.

Reply:
> It should be always present (if needed at all), otherwise this can cause crashes when checking the value.

Where for example? And how to restore the original / default reference density value?

> It'll also cause issues when serializing/deserializing objects.

Where for example?